File: mobile/lib/shared/brand/zion_release.dart
enum ZionReleaseChannel { developer, release }

// Zion stays visibly pre-release unless the build explicitly passes
// --dart-define=ZION_RELEASE_CHANNEL=release.
const _configuredZionReleaseChannel = String.fromEnvironment(
  'ZION_RELEASE_CHANNEL',
  defaultValue: 'developer',
);

ZionReleaseChannel resolveZionReleaseChannel(String? value) {
  return value == ZionReleaseChannel.release.name
      ? ZionReleaseChannel.release
      : ZionReleaseChannel.developer;
}

final currentZionReleaseChannel = resolveZionReleaseChannel(
  _configuredZionReleaseChannel,
);

String formatZionReleaseLabel(String version, ZionReleaseChannel channel) {
  final developerSuffix = channel == ZionReleaseChannel.developer ? ' DV' : '';
  return 'Zion - V$version$developerSuffix';
}

File: desktop/src/shared/constants/zionRelease.ts
export const ZionReleaseChannel = {
  Developer: "developer",
  Release: "release",
} as const;

export type ZionReleaseChannel =
  (typeof ZionReleaseChannel)[keyof typeof ZionReleaseChannel];

export type ZionBuildInfo = {
  buildId: string;
  commit: string;
  worktree: "clean" | "dirty" | "unknown";
};

export function resolveZionReleaseChannel(
  value: string | undefined,
): ZionReleaseChannel {
  return value === ZionReleaseChannel.Release
    ? ZionReleaseChannel.Release
    : ZionReleaseChannel.Developer;
}

// Zion stays visibly pre-release unless a production build explicitly sets
// VITE_ZION_RELEASE_CHANNEL=release.
export const currentZionReleaseChannel = resolveZionReleaseChannel(
  import.meta.env?.VITE_ZION_RELEASE_CHANNEL,
);

function readBuildEnv(key: string, fallback: string): string {
  const value = import.meta.env?.[key]?.trim();
  return value || fallback;
}

export const currentZionBuild: ZionBuildInfo = {
  buildId: readBuildEnv("VITE_ZION_BUILD_ID", "local"),
  commit: readBuildEnv("VITE_ZION_COMMIT", "unknown"),
  worktree: (() => {
    const value = readBuildEnv("VITE_ZION_WORKTREE", "unknown");
    return value === "clean" || value === "dirty" ? value : "unknown";
  })(),
};

export function formatZionReleaseLabel(
  version: string,
  channel: ZionReleaseChannel,
): string {
  const developerSuffix = channel === ZionReleaseChannel.Developer ? " DV" : "";
  return `Zion - V${version}${developerSuffix}`;
}

export function formatZionBuildLabel(build: ZionBuildInfo): string {
  const worktreeSuffix = build.worktree === "dirty" ? " · local changes" : "";
  return `Build ${build.buildId} · commit ${build.commit}${worktreeSuffix}`;
}
